### ISSUE TITLE
no-unused-vars rule don't work while extending the airbnb's config

### ISSUE THREAD

**Issue by artptr** [eslint/eslint#5462]
[This repo](https://github.com/artptr/eslint-airbnb-no-unused-vars-bug) reproduces the bug in `eslint-config-airbnb`.
There is an error while checking the `test.js`:

```
  1:5  error  'Foo' is defined but never used  no-unused-vars
```

The error appears despite that the parent config contains appropriate `no-unused-vars` setting.
It disappears only if we remove [the airbnb's config](https://github.com/airbnb/javascript/tree/master/packages/eslint-config-airbnb) (default or legacy) from the `extends` property.

The error appears on the ESLint 1.10.3 and 2.2.0

This issue is the copy of https://github.com/airbnb/javascript/issues/773

**Comment by gyandeeps** [comment]
Can you please share what you expect to happen?

**Comment by ljharb** [comment]
Per https://github.com/airbnb/javascript/issues/773#issuecomment-191884169, "According to the eslint docs, setting the "vars" option should still warn about unused local vars - it seems that error is correct, and that var is unused, meaning that line could be deleted and your program would work the same."

**Comment by ljharb** [comment]
@gyandeeps this can be closed - it's just that our env is set to "node" and the user expected that a "legacy" config would be set to "browser".

**Comment by artptr** [comment]
@ljharb Of cource and thank you for your participation!
